app/(interactive)/4-1/page.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+import { useRouter } from "next/navigation";
+import { getStroke } from "perfect-freehand";
+import React, { useContext, useRef } from "react";
+import AppContext from "../../../components/AppContext";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  function getSvgPathFromStroke(stroke: any) {
+    if (!stroke.length) return "";
+    const d = stroke.reduce(
+      (acc: any[], [x0, y0]: any, i: number, arr: string | any[]) => {
+        const [x1, y1] = arr[(i + 1) % arr.length];
+        acc.push(x0, y0, (x0 + x1) / 2, (y0 + y1) / 2);
+        return acc;
+      },
+      ["M", ...stroke[0], "Q"],
+    );
+    d.push("Z");
+    return d.join(" ");
+  }
+
+  const [hasWrite, setWrite] = React.useState<boolean>(false);
+  const [strokes, setStrokes] = React.useState<number[][][]>([]);
+  const currentStroke = useRef<number[][]>([]);
+  const context = useContext(AppContext);
+  const router = useRouter();
+
+  function handlePointerDown(e: React.PointerEvent<SVGSVGElement>) {
+    setWrite(true);
+    e.currentTarget.setPointerCapture(e.pointerId);
+    var frame = document.getElementById("svg");
```

Comment:
try not to use `document.` functions kub since React works a bit differently from how HTML DOM works. Any `document.` functions can be written to be `ref` items from `useRef`.

How to use:
1) Create a ref state
```ts
const svgRef = useRef();
```

2) Let an element consume ref
```tsx
<SVG ref={svgRef} />
```

3) Use item's element
```ts
const pos = svgRef?.current.getBoundingClientRect();
```